lib/ansible/inventory/data.py

```diff
@@ -234,8 +234,10 @@ def add_host(self, host, group=None, port=None):
 
     def remove_host(self, host):
 
-        if host.name in self.hosts:
-            del self.hosts[host.name]
+        for hostobj in self.hosts:
+            # match on host name/inventory_hostname is unique id
+            if hostobj.name == host.name:
+                del self.hosts[hostobj]
```

Comment:
Wouldn't this fail with `RuntimeError: dictionary changed size during iteration`?

Reply:
it breaks, should never hit the 'next iteration'